tinygrad/codegen/linearizer.py

```diff
@@ -412,7 +412,11 @@ def uop(self, uop:UOps, dtype:Optional[DType], vin:Tuple[UOp, ...], arg:Any=None
   def ast_parse(self, x, acc, offs, loaded_buffers, do_reduce=False) -> List[UOp]:
     if x.__class__ is not LazyOp: return loaded_buffers[x]    # for LOCAL_BUFFER
     if x.op in BufferOps: return loaded_buffers[x.arg]
-    if x.op in [UnaryOps.NOOP, UnaryOps.CAST]: return self.ast_parse(x.src[0], acc, offs, loaded_buffers)  # cast isn't an ALU op
+    if x.op == UnaryOps.NOOP: return self.ast_parse(x.src[0], acc, offs, loaded_buffers)
+    if x.op == UnaryOps.CAST:
+      dtype, bitcast = x.arg
+      if not bitcast: return self.ast_parse(x.src[0], acc, offs, loaded_buffers)
+      return [self.uop(val.uop, dtype, val.vin, (val.arg, bitcast) if val.uop in [UOps.GEP, UOps.LOAD] else val.arg) for val in self.ast_parse(x.src[0], acc, offs, loaded_buffers)]
```

Comment:
This is sort of an unmergable hack. I think this might be blocked on:
"Non FP32 math, internal casting support, well tested"
Upside, it's another $500.

Reply:
unblocked now https://github.com/tinygrad/tinygrad/commit/4380ccb1699e32849819f2c043c15579b160de39